src/app/models/constraint-models/combination-constraint.ts

```diff
@@ -10,17 +10,16 @@ import {Constraint} from './constraint';
 import {CombinationState} from './combination-state';
 import {PedigreeConstraint} from './pedigree-constraint';
 import {TrueConstraint} from './true-constraint';
-import {ConstraintMark} from './constraint-mark';
 
 export class CombinationConstraint extends Constraint {
-
   private _children: Constraint[];
   private _combinationState: CombinationState;
   private _isRoot: boolean;
+  private _dimension: string;
 
   constructor(children?: Constraint[],
               state?: CombinationState,
-              mark?: ConstraintMark) {
+              dimension?: string) {
```

Comment:
still think we should make this enum, to serve as a gate guarding what kinds of dimensions are allowed in a combination constraint. Enum is meant to deal with pre-determined / hardcoded scenarios.

Reply:
Actually dimensions depend on the loaded data, so I don't thing hardcoding them would be a good option, e.g. they can be named differently depending on how they were loaded ("diagnosis", "Diagnosis ID" etc). "patient" dimension is the only one that should be always the same.

Reply:
wait, these dimensions are used as modifiers in transmart db, how can data loaders decide the names of dimensions? shouldn't transmart db already have guidelines for that?

Reply:
in transmart the dimension names end up in two tables - observations and modifier description table, but the name can be anything - they are not hardcoded in the transmart code.

Reply:
ok, I see